tests/infra-validations.sh

```diff
@@ -100,3 +100,16 @@ file_we_expect=("definition.yaml")
 validate_directory "$TEST_WORKSPACE/$infra_hld_dir" "${file_we_expect[@]}"
 # Validate the contents of the definition.yaml
 validate_file "$TEST_WORKSPACE/$infra_hld_dir/definition.yaml" $validation_test_yaml >> $TEST_WORKSPACE/log.txt
+
+# Generate TF Files from Infra HLDs ------------------
+cd $infra_hld_dir
+spk infra generate
+
+# Verify that the Terraform files generation was successful
+# Confirm that generated directory created, spk.tfvars created, and tf templates copied
+generated_directory="$TEST_WORKSPACE/$infra_hld_dir-generated"
+file_we_expect=("spk.tfvars" "main.tf" "variables.tf")
+validate_directory "$generated_directory" "${file_we_expect[@]}"
+
+# Confirm contents of the spk.tfvars file are correct
+validate_file "$generated_directory/spk.tfvars" 'rg_name = "<insert value>"'
```

Comment:
@yradsmikham One caveat I see is that we're not testing for backend or partial config on the generate at 2 levels instead of 1.

Reply:
Not a problem, we can add backend configuration to the yaml file and test for it.